버그수정, docs 추가

maong0927 opened the pull request:
### 변경사항
1. 노드 버전에서만 errorContext 받아올 수 있도록 변경
2. 디버깅용 console.log 제거 
3. node, browser npm 문서 / 추가된 함수,  option들 추가
4. core, utils, types npm 문서 / 로고, 깃허브 주소 제목 추가 


### 작업 유형
- [ ] 신규 기능 추가
- [x] 버그 수정
- [x] 리펙토링
- [x] 문서 업데이트

### 체크리스트
- [x] Merge 하는 브랜치가 올바른가?
- [x] 코딩컨벤션을 준수하는가?
- [x] PR과 관련없는 변경사항이 없는가?
- [x] 내 코드에 대한 자기 검토가 되었는가?
- [ ] 변경사항이 효과적이거나 동작이 작동한다는 것을 보증하는 테스트를 추가하였는가?
- [ ] 새로운 테스트와 기존의 테스트가 변경사항에 대해 만족하는가?

maong0927 committed:
feat    : example 코드 수정

1. unhandleRejectionLevel
2. uncaughtExceptionLevel
옵션 추가

maong0927 committed:
Publish

 - @santry/browser@1.2.0
 - @santry/core@1.3.0
 - @santry/node@1.3.0
 - @santry/types@1.3.0
 - @santry/utils@1.3.0

maong0927 committed:
Publish

 - @santry/browser@1.3.0
 - @santry/core@1.4.0
 - @santry/node@1.4.0
 - @santry/types@1.4.0
 - @santry/utils@1.4.0

maong0927 committed:
fix     : errorStackparsing 브라우저 버전에서 제거

1. 브라우저 버전에서는 fs 모듈을 불러올 수 없어 errorStack parsing 제거

maong0927 committed:
feat    : error type, value 만 가져오는 함수 구현

maong0927 committed:
fix     : errorStack parse 만 할 수 있도록 수정

maong0927 committed:
feat    : getErrorInfo 함수 수정

maong0927 committed:
fix     : browser 에서는 fs 로 스택 parse 하지 않도록 수정

maong0927 committed:
fix     : 버그 수정

1. define 되지않은 property 에 값을넣어서 에러가뜸

maong0927 committed:
Publish

 - @santry/browser@1.3.1
 - @santry/core@1.4.1
 - @santry/node@1.4.1
 - @santry/utils@1.4.1

maong0927 committed:
fix     : index.ts 삭제

maong0927 committed:
Publish

 - @santry/browser@1.3.2
 - @santry/core@1.4.2
 - @santry/node@1.4.2
 - @santry/utils@1.4.2

maong0927 committed:
fix     : fs node 버전에서 주입하도록 변경

maong0927 committed:
Publish

 - @santry/browser@1.3.3
 - @santry/core@1.4.3
 - @santry/node@1.4.3
 - @santry/utils@1.4.3

maong0927 committed:
refactor: errorStack을 node, browser 공통으로 설정함

maong0927 committed:
Publish

 - @santry/browser@1.3.4
 - @santry/core@1.4.4
 - @santry/node@1.4.4
 - @santry/utils@1.4.4

maong0927 committed:
fix     : error 정보가 두번 들어가던 문제 해결

maong0927 committed:
refactor: 기능 변경으로 인한 메소드 이름 변경

1. 기존 ErrorStack을 parse 해서 error 관련 정보들을 return 하는 함수 였지만
2. parse 한 뒤 errorContext 만 return 해주게 바뀌어서 getErrorContext로 이름 변경

maong0927 committed:
Publish

 - @santry/browser@1.3.5
 - @santry/core@1.4.5
 - @santry/node@1.4.5
 - @santry/utils@1.4.5

maong0927 committed:
docs    : example 코드 수정

1. 함수 이름 수정
2. console.log 제거

maong0927 committed:
docs    : node 버전 npm 문서 수정

1. 새로 추가된 options 추가
2. 주석, 설명 더 추가
3. setLevel 없어진 것 반영

maong0927 committed:
docs    : browser 버전 npm 문서 수정

1. options 추가

maong0927 committed:
docs    : sdk 와 관련없는 기술 스택 제거

maong0927 committed:
refactor: 안쓰는 console.log 제거

maong0927 committed:
docs    : core, utils, types npm 문서 추가

1. 로고, 깃허브, 제목 추가

maong0927 committed:
Publish

 - @santry/browser@1.3.6
 - @santry/core@1.4.6
 - @santry/node@1.4.6
 - @santry/types@1.4.1
 - @santry/utils@1.4.6

maong0927 committed:
docs    : 필요없는 구분선 제거

maong0927 committed:
Publish

 - @santry/browser@1.3.7
 - @santry/core@1.4.7
 - @santry/node@1.4.7
 - @santry/types@1.4.2
 - @santry/utils@1.4.7

maong0927 committed:
refactor: Santry interface implement 추가

1. baseSantry가 Santry interface 를 implement 하도록 변경
2. 쓰임에 따라 protected, public, private 수정

maong0927 committed:
Publish

 - @santry/browser@1.3.8
 - @santry/core@1.4.8
 - @santry/node@1.4.8
 - @santry/types@1.4.3
 - @santry/utils@1.4.8

gitdog01 commented on examples/node.example.js:
```diff
@@ -1,20 +1,16 @@
   traceSampleRate: 1,
   release: 'santry@0.0.1',
   environment: 'production',
+  unhandledRejectionLevel: 'critical',
+  uncaughtExceptionLevel: 'warning',
```
그냥 Error 와 다른 ErrorLevel 을 적용했군요.

maong0927 replied:
네 ㅎㅎㅎ맞습니당 ~~

gitdog01 commented on examples/node.example.js:
```diff
@@ -25,8 +21,8 @@ const testError = () => {
 
-const testUnhandledRejection = () => {
+/* const testUnhandledRejection = () => {
   this.loaded = Promise.reject(new Error('Resource not yet loaded!'));
 };
 
-const resource = testUnhandledRejection();
+const resource = testUnhandledRejection(); */
```
주석처리 무엇?..

maong0927 replied:
헐 고치겠습니당!

gitdog01 commented on packages/core/src/baseSantry.ts:
```diff
@@ -66,7 +67,7 @@ export abstract class BaseSantry {
 
     // Error 정보
     else {
-      event.error = parseErrorStack(content);
+      event.error = { ...event.error, ...getErrorInfo(content) };
```
정상작동하나요 ?? Parse 해주는것이 있다가 변경되었는데

maong0927 replied:
정상작동합니당~

gitdog01 commented on packages/utils/src/getErrorInfo.ts:
```diff
@@ -0,0 +1,34 @@
+    stacktrace?: StackTrace[];
+  } = {};
+  const parsedStackList = parse(error);
+  if (parsedStackList) {
+    event.stacktrace = parsedStackList.map((stack) => {
+      try {
+        String(stack).replace(/(\\r\\n|\\n|\\r)/gm, '\\n');
+        return {
+          filename: stack.fileName,
+          function: stack.functionName,
+          lineno: stack.lineNumber,
+          colno: stack.columnNumber,
+        };
+      } catch {
+        return {
+          filename: stack.fileName,
+          function: stack.functionName,
+          lineno: stack.lineNumber,
+          colno: stack.columnNumber,
+        };
+      }
```
제대로 작동하는지 확인해봐야할 것 같아요

maong0927 replied:
정상작동해용~

gitdog01 reviewed (approved):
고생하셨어요~

minbert reviewed (approved):
수고하셨습니다.

maong0927 committed:
feat    : example에 unhandledRejection test 추가

maong0927 committed:
chore   : 안쓰는 설정 삭제

1. jest 관련 설정 삭제

maong0927 committed:
chore   : 안쓰는 폴더 삭제

maong0927 committed:
Publish

 - @santry/browser@1.3.9
 - @santry/core@1.4.9
 - @santry/node@1.4.9
 - @santry/types@1.4.4
 - @santry/utils@1.4.9

Eunbin-Kim commented on packages/utils/src/getErrorContext.ts:
```diff
@@ -1,25 +1,13 @@
 import { parse } from 'error-stack-parser';
 
-export const parseErrorStack = (error: Error): any => {
+export const getErrorContext = (fs: any, error: Error): any => {
```
fs를 주입했다는게 parameter로 설정한것이군여

maong0927 replied:
넹넹 맞아용!!

Eunbin-Kim reviewed (approved):
고생 많으셨습니당!
